### ISSUE TITLE
Problem with generated keys

### ISSUE THREAD

**Issue by obraliar** [impossibl/pgjdbc-ng#283]
I try to get an auto generated key from a field which contains uppercase characters:

```
PreparedStatement preparedEntryStatement = connection.prepareStatement(query, new String[] {"ID"})
```

I wrote a small test case (https://github.com/obraliar/pgjdbc-ng-test/blob/master/src/test/java/KeyGeneratorTest.java#L42), which shows following result [on travis](https://travis-ci.org/obraliar/pgjdbc-ng-test/builds/162052139#L276):

```
KeyGeneratorTest > insert[Test with com.impossibl.postgres.jdbc.PGConnectionImpl@6718a012] FAILED
    com.impossibl.postgres.jdbc.PGSQLSimpleException: column "id" does not exist
        at com.impossibl.postgres.jdbc.ErrorUtils.makeSQLException(ErrorUtils.java:159)
        at com.impossibl.postgres.jdbc.ErrorUtils.makeSQLException(ErrorUtils.java:136)
        at com.impossibl.postgres.jdbc.PGConnectionImpl.execute(PGConnectionImpl.java:442)
        at com.impossibl.postgres.jdbc.PGPreparedStatement$1.call(PGPreparedStatement.java:206)
        at com.impossibl.postgres.jdbc.PGPreparedStatement$1.call(PGPreparedStatement.java:196)
        at com.impossibl.postgres.jdbc.PGConnectionImpl.getCachedStatement(PGConnectionImpl.java:1301)
        at com.impossibl.postgres.jdbc.PGPreparedStatement.parseIfNeeded(PGPreparedStatement.java:196)
        at com.impossibl.postgres.jdbc.PGPreparedStatement.execute(PGPreparedStatement.java:262)
        at com.impossibl.postgres.jdbc.PGPreparedStatement.executeUpdate(PGPreparedStatement.java:291)
        at KeyGeneratorTest.insert(KeyGeneratorTest.java:43)
```

This statement is fully working with the  `org.postgresql:postgresql:9.4.1209`:

```
KeyGeneratorTest > insert[Test with org.postgresql.jdbc.PgConnection@ec9f8e6] PASSED
```

I'm using the `com.impossibl.pgjdbc-ng:pgjdbc-ng:0.6` driver.

**Comment by kdubb** [comment]
Looks like the fork is gone any chance we can get that test back? Apologies, I know this is an __old__ issue.

**Comment by kdubb** [comment]
Reproduced, fixed & test case added in c23acc639163b6e16d90f49fce5441e6c1b4561c
